app/connectors/parsers/getPenaltyDetails/HIPPenaltyDetailsParser.scala

```diff
@@ -0,0 +1,166 @@
+
+  implicit object HIPPenaltyDetailsReads extends HttpReads[HIPPenaltyDetailsResponse] {
+    override def read(method: String, url: String, response: HttpResponse): HIPPenaltyDetailsResponse = {
+      response.status match {
+        case OK =>
+          logger.debug(s"[HIPPenaltyDetailsReads][read] Json response: ${response.json}")
+          response.json.validate[PenaltyDetails] match {
+            case JsSuccess(getPenaltyDetails, _) =>
+              logger.debug(s"[HIPPenaltyDetailsReads][read] Model: $getPenaltyDetails")
+              Right(HIPPenaltyDetailsSuccessResponse(addMissingLPP1PrincipalChargeLatestClearing(getPenaltyDetails)))
+            case JsError(errors) =>
+              logger.debug(s"[HIPPenaltyDetailsReads][read] Json validation errors: $errors")
+              Left(HIPPenaltyDetailsMalformed)
+          }
+        case NOT_FOUND if response.body.nonEmpty => {
+          Try(handleNotFoundStatusBody(response.json)).fold(parseError => {
+            logger.error(s"[PenaltyDetailsReads][read] Could not parse 404 body with error ${parseError.getMessage}")
+            PagerDutyHelper.log("PenaltyDetailsReads", INVALID_JSON_RECEIVED_FROM_1812_API)
+            Left(HIPPenaltyDetailsFailureResponse(NOT_FOUND))
+          }, identity)
+        }
+          case status @ (BAD_REQUEST | CONFLICT | INTERNAL_SERVER_ERROR |
+            SERVICE_UNAVAILABLE | UNPROCESSABLE_ENTITY) => {
+
+          val json = response.json
+
+          val maybeError: Option[PenaltyApiError] =
+            (json \ "error")
+              .validate[TechnicalError]
+              .asOpt
+              .orElse(
+                (json \ "errors").validate[BusinessError].asOpt
+              )
+
+          maybeError.foreach {
+            case te: TechnicalError =>
+              logger.warn(
+                s"[HIPPenaltyDetailsReads] Technical error: ${te.code} / ${te.logID}"
+              )
+            case be: BusinessError =>
+              logger.warn(
+                s"[HIPPenaltyDetailsReads] Business error: ${be.code} / ${be.text}"
+              )
+          }
+
+          Left(HIPPenaltyDetailsFailureResponse(status, maybeError))
+        }
+        case status@NO_CONTENT => {
+          logger.debug(s"[HIPPenaltyDetailsReads][read] Received 204 when calling ETMP")
+          Left(HIPPenaltyDetailsFailureResponse(status))
+        }
+        case _@status =>
+          PagerDutyHelper.logStatusCode("HIPPenaltyDetailsReads", status)(RECEIVED_4XX_FROM_1812_API, RECEIVED_5XX_FROM_1812_API)
+          logger.error(s"[HIPPenaltyDetailsReads][read] Received unexpected response from GetPenaltyDetails, status code: $status and body: ${response.body}")
+          Left(HIPPenaltyDetailsFailureResponse(status))
+      }
+    }
+  }
+
+  private def handleNotFoundStatusBody(responseBody: JsValue): HIPPenaltyDetailsResponse = {
+    (responseBody \ "failures").validate[Seq[FailureResponse]].fold(
```

Comment:
Do we know that NotFound will have this error format?
It doesn't show an example in the schema